tools/mo/openvino/tools/mo/ops/Cast.py

```diff
@@ -83,10 +89,12 @@ def custom_type_casting_and_packing(node: Node, value, dst_type):
         padded = np.concatenate((flattened, np.zeros([pad], dtype=minimum_regular_dtype)))
         assert np.prod(padded.shape) % num_values_fitting_into_uint8 == 0
 
-        bit_order_little = (padded[:, None] & (1 << np.arange(num_bits)) > 0).astype(np.uint8)
-        bit_order_big = np.flip(bit_order_little, axis=1)
-        bit_order_big_flattened = bit_order_big.flatten()
-        packed = np.packbits(bit_order_big_flattened)
+        bit_order_little = (padded[:, None] & (
+            1 << np.arange(num_bits)) > 0).astype(np.uint8)
+        bit_order_big_flattened = bit_order_little.flatten()
+        # u1 still has reversed bit order:
+        packed = np.packbits(bit_order_big_flattened,
+                             bitorder='little' if num_bits > 1 else 'big')
```

Comment:
num_bits always == 4 for this function

Reply:
Good point, but the code around was written in a way that `num_bits` was treated as a parameter (look at `np.arange(num_bits)`) so I have just left the same "portable" style.